File: maximum-subarray_53.js
const arr = [-2, 1, -3, 4, -1, 2, 1, -5, 4];

const maxSubArray = (nums) => {
  let maxSubArr = nums[0];
  let currSum = 0;

  for (let i = 0; i < nums.length; i++) {
    if (currSum < 0) currSum = 0;
    currSum += nums[i];
    maxSubArr = Math.max(maxSubArr, currSum);
  }

  return maxSubArr;
};

const result = maxSubArray(arr);

console.log(result);

File: maximum-subarray_53.py
nums = [-2, 1, -3, 4, -1, 2, 1, -5, 4]


def maxSubArray(nums):
    if len(nums) == 0:
        return 0
        
    max_sub = nums[0]
    print(max_sub)
    cur_sum = 0

    for num in nums:
        if cur_sum < 0:
            cur_sum = 0

        cur_sum += num
        max_sub = max(max_sub, cur_sum)

    return max_sub


output = maxSubArray(nums)
# print(output)
